src/pages/RepoPage/BundlesTab/BundleContent/BundleSummary/BundleSelector.spec.tsx

```diff
@@ -0,0 +1,297 @@
+            message: 'missing head report',
+          },
+        },
+      },
+    },
+  },
+}
+
+const server = setupServer()
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      retry: false,
+      suspense: true,
+    },
+  },
+})
+
+let testLocation: ReturnType<typeof useLocation>
+const wrapper =
+  (
+    initialEntries = '/gh/codecov/test-repo/bundles/test-branch'
+  ): React.FC<React.PropsWithChildren> =>
+  ({ children }) =>
+    (
+      <QueryClientProvider client={queryClient}>
+        <MemoryRouter initialEntries={[initialEntries]}>
+          <Route
+            path={[
+              '/:provider/:owner/:repo/bundles/:branch/:bundle',
+              '/:provider/:owner/:repo/bundles/:branch',
+              '/:provider/:owner/:repo/bundles/',
+            ]}
+          >
+            <Suspense fallback={<p>Loading</p>}>{children}</Suspense>
+          </Route>
+          <Route
+            path="*"
+            render={({ location }) => {
+              testLocation = location
+              return null
+            }}
+          />
+        </MemoryRouter>
+      </QueryClientProvider>
+    )
+
+beforeAll(() => {
+  server.listen()
+})
+
+afterEach(() => {
+  queryClient.clear()
+  server.resetHandlers()
+})
+
+afterAll(() => {
+  server.close()
+})
+
+interface SetupArgs {
+  missingHeadReport?: boolean
+  nullOverview?: boolean
+}
+
+describe('BundleSelector', () => {
+  function setup({
+    missingHeadReport = false,
+    nullOverview = false,
+  }: SetupArgs) {
+    const user = userEvent.setup()
+
+    server.use(
+      graphql.query('GetRepoOverview', (req, res, ctx) => {
+        if (nullOverview) {
+          return res(ctx.status(200), ctx.data({ owner: null }))
+        }
+
+        return res(
+          ctx.status(200),
+          ctx.data({ owner: { repository: mockRepoOverview } })
+        )
+      }),
+      graphql.query('BranchBundlesNames', (req, res, ctx) => {
+        if (missingHeadReport) {
+          return res(ctx.status(200), ctx.data(mockBadBundles))
+        } else return res(ctx.status(200), ctx.data(mockBundles))
+      })
+    )
+
+    return { user }
+  }
+
+  describe('when there are no bundles', () => {
+    it('disables the select', async () => {
+      setup({ missingHeadReport: true })
+      render(<BundleSelector />, { wrapper: wrapper() })
+
+      const select = await screen.findByRole('button', {
+        name: 'bundle tab bundle selector',
+      })
+      expect(select).toBeInTheDocument()
+      expect(select).toBeDisabled()
+    })
+  })
+
+  describe('when there is no overview data', () => {
+    it('renders disabled button', async () => {
+      setup({ nullOverview: true })
+      render(<BundleSelector />, {
+        wrapper: wrapper('/gh/codecov/test-repo/bundles'),
+      })
+
+      const select = await screen.findByRole('button', {
+        name: 'bundle tab bundle selector',
+      })
+      expect(select).toBeInTheDocument()
```

Comment:
Shouldn't there be a `expect(select).toBeDisabled()` here?

Reply:
whoops, meant to delete that test